test-common/Nettest/WhitelistIntegration.hs

```diff
@@ -0,0 +1,94 @@
+-- SPDX-FileCopyrightText: 2020 TBD
+--
+-- SPDX-License-Identifier: LicenseRef-Proprietary
+module Nettest.WhitelistIntegration
+  ( whitelistScenario
+  ) where
+
+import qualified Data.Map as Map (fromList)
+import qualified Data.Set as Set (fromList)
+
+import Lorentz hiding (sender)
+import Michelson.Typed.Convert (untypeValue)
+import Michelson.Typed.Haskell.Value (IsoValue(..))
+import qualified Michelson.Untyped as U
+import Morley.Nettest
+import Morley.Nettest.Caps (originateUntypedSimple)
+import Util.Named
+
+import Indigo.Contracts.Holdings
+
+type WhitelistId = Natural
+
+data WhitelistStorage = WhitelistStorage
+  { wsIssuer :: Address
+  , wsUsers :: BigMap Address WhitelistId
+  , wsWhitelists :: BigMap WhitelistId (Bool, Set WhitelistId)
+  , wsAdmin :: Address
+  }
+  deriving stock Generic
+  deriving anyclass IsoValue
+
+mkWhitelistStorage
+  :: Address -> [(Address, WhitelistId)] -> [(WhitelistId, (Bool, Set WhitelistId))]
+  -> WhitelistStorage
+mkWhitelistStorage issuer users whitelists = WhitelistStorage issuer
+  (BigMap $ Map.fromList users) (BigMap $ Map.fromList whitelists) issuer
```

Comment:
Can you please use record syntax here? This way it will be clearer that the second `issuer` is `wsAdmin`.